### PR TITLE
ghc: add page

### PR DESCRIPTION
**Pull request by alx741**
The Glasgow **Haskell compiler**
https://www.haskell.org/ghc/

### CODE REVIEW

File: pages/common/ghc.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ghc
+
+> The Glassgow Haskell Compiler, compiles and links Haskell source files.
```
Comment (waldyrious) [inline pages/common/ghc.md]: maybe use a period, dash, semicolon or colon (your preference) instead of a comma here.

File: pages/common/ghc.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+- Find and compile all modules in the current directory:
+
+`ghc Main`
+
+- Compile a single file:
+
+`ghc {{file.hs}}`
+
+- Compile using optimization, for faster code:
```
Comment (waldyrious) [inline pages/common/ghc.md]: I would say the "for faster code" is unnecessary here, and doesn't match the similar entries for other compilers. If we decide to keep this, then the others should be edited to say the same.

File: pages/common/ghc.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+`ghc -O {{file.hs}}`
+
+- Stop after generating object files:
```
Comment (waldyrious) [inline pages/common/ghc.md]: Stop what? It would be nice to be a tad more descriptive here, for total beginners.

File: pages/common/ghc.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+`ghc -c {{file.hs}}`
+
+- Run in interactive mode (REPL):
```
Comment (waldyrious) [inline pages/common/ghc.md]: Maybe "run the Haskell interpreter" or something to that effect?

**Comment by waldyrious** [comment]
Another great one, thanks. I've left a few inline comments.

File: pages/common/ghc.md (hunk 4 above)
Comment (alx741) [inline pages/common/ghc.md]: The reason is that the documentation calls it *"the interactive mode"* while the *"Haskell interpreter"* is actually a complete separate program. But I think i could try something else

**Comment by alx741** [comment]
Thank you for the suggestions!
Let me know if there is something else to be improved

**Comment by waldyrious** [comment]
No further issues, LGTM as it is! 👍 from me.

**Pull request merged**